unblob/pool.py

```diff
@@ -157,3 +197,20 @@ def make_pool(process_num, handler, result_callback) -> Union[SinglePool, MultiP
         handler=handler,
         result_callback=result_callback,
     )
+
+
+orig_signal_handlers = {}
+
+
+def _on_terminate(signum, frame):
+    with contextlib.suppress(StopIteration):
+        while True:
+            pool = next(iter(pools))
+            pool.close(immediate=True)
```

Comment:
Maybe this way so `if callable...` is reachable:

```suggestion
        for pool in pools:
            pool.close(immediate=True)
```

Reply:
not really, as close mutates the pools set

Reply:
How about this ?

```py
while pool := next(iter(pools)):
    pool.close(immediate=True)
```

Reply:
It will never be false anyway as `next` will return something or raise an error.